xdsl/transforms/eqsat_optimize_pdl_interp.py

```diff
@@ -0,0 +1,154 @@
+from dataclasses import dataclass, field
+
+from xdsl.context import Context
+from xdsl.dialects import builtin, pdl_interp
+from xdsl.ir import Block, SSAValue, Use
+from xdsl.passes import ModulePass
+from xdsl.traits import SymbolTable
+
+
+@dataclass(frozen=True)
+class EqsatOptimizePDLInterp(ModulePass):
+    name = "eqsat-optimize-pdl-interp"
+
+    def apply(self, ctx: Context, op: builtin.ModuleOp) -> None:
+        matcher = SymbolTable.lookup_symbol(op, "matcher")
+        assert isinstance(matcher, pdl_interp.FuncOp)
+        assert matcher is not None, "matcher function not found"
+
+        mo = MatcherOptimizer(matcher)
+        mo.optimize()
```

Comment:
this seems to be unused? Do we want to expose this as a pass? If yes, can you please add a test?